CONTRIBUTING.md

```diff
@@ -225,9 +225,30 @@ build all the documentation into `docs/_build/html` and the release notes in
 particular will be located at `docs/_build/html/release_notes.html`
 
 ## Installing Qiskit Terra from source
-Please see the [Installing Qiskit Terra from
-Source](https://qiskit.org/documentation/contributing_to_qiskit.html#installing-terra-from-source)
-section of the Qiskit documentation.
+
+Qiskit Terra is primarily written in Python but there are some core routines
+that are written in the [Rust](https://www.rust-lang.org/) programming
+language to improve the runtime performance. For the released versions of
+qiskit-terra we publish precompiled binaries on the
+[Python Package Index](https://pypi.org/) for all the supported platforms
+which only requires a functional Python environment to install. However, when
+building and installing from source you will need a rust compiler installed
+to be able to build and install qiskit from source. You can do this very easily
+using rustup: https://rustup.rs/ which provides a single tool to install and
+configure the latest version of the rust compiler.
+[Other installation methods](https://forge.rust-lang.org/infra/other-installation-methods.html)
+exist too. For windows users besides rustup you will also need install
+the Visual C++ build tools so that rust can link against the system c/c++
+libraries. You can see more details on this in the rustup documentation:
+
+https://rust-lang.github.io/rustup/installation/windows.html
+
+Once you have a rust compiler installed you can rely on the normal Python
+build/install steps to install Qiskit Terra. This means you just run
+`pip install .` in your local git clone to build and install Qiskit Terra.
+
+Do note that if you do use develop mode/edittable install (via `python setup.py develop` or `pip install -e .`) The rust extension will be built in debug mode
+without any optimizations enabled. This will result in poor runtime performance.
```

Comment:
There might be, I'd have to check the cargo docs and setuptools-rust (to ensure it passes any flags through). One option you can always do is `python setup.py build_rust --release --inplace && pip install -e .` that will build the extension in release mode but also setup the python packaging link.

Reply:
The other trick I often do for local quick iteration is to just bypass python/pip and just running `cargo build --release && cp target/release/libqiskit.so qiskit/_accelerate*.so` (where `accelerate*.so` is the filename that exists with your python version and environment locally) So, you could do `pip install -e .` and follow it up with the cargo command and get the same effect

Reply:
Oh yeah, I didn't want to remove it by default by any stretch, I was just wondering if there was a way around it, because 99% of the time I'm running Qiskit it's with editable installs.